src/sinol_make/commands/run/__init__.py

```diff
@@ -360,78 +367,184 @@ def print_view(output_file=None):
-			programs.append(validator_program)
-		programs = list(set(programs))
-		compilation_results = self.compile_programs(programs)
-		os.makedirs(self.EXECUTIONS_DIR, exist_ok=True)
-		compiled_commands = []
-		for subtask in self.config["subtasks"]:
-			validator_program = os.path.join(self.EXECUTABLES_DIR, self.config["subtasks"][subtask]["validator"])
-			compiled_commands.append((subtask, validator_program, True))
-		names = list(self.config["subtasks"])
-		results = self.perform_executions(compiled_commands, names, programs, self.args.subtask_report)
-		for subtask in self.config["subtasks"]:
-			passed_groups = []
-			for group in self.config["scores"]:
-				if group in results[subtask] and results[subtask][group] == "OK":
-					passed_groups.append(group)
-			passed_groups.sort()
-			should_pass = self.config["subtasks"][subtask]["groups"]
-			should_pass.sort()
-			if passed_groups != should_pass:
-				print(util.bold(util.color_red('Subtask %s will pass groups %s (expected %s).' % (subtask, passed_groups, should_pass))))
-				exit(1)
-
-
-	def run_programs(self):
-		programs = self.get_programs(self.args.programs)
-		print("The following %d programs will be executed:\n%s"
-			% (len(programs), [self.extract_program_name(program) for program in programs]))
-		print("on the following %d tests:\n%s"
-			% (len(self.tests), [self.extract_test_no(test) for test in self.tests] ))
-		print("in parallel on %d cpus." % self.cpus)
-		print()
-		compilation_results = self.compile_programs(programs)
+
+	def run_solutions(self, solutions):
+		compilation_results = self.compile_solutions(solutions)
 		os.makedirs(self.EXECUTIONS_DIR, exist_ok=True)
-		program_executables = [os.path.join(self.EXECUTABLES_DIR, self.get_executable(program))
-							for program in programs]
-		compiled_commands = zip(programs, program_executables, compilation_results)
-		names = programs
-		self.perform_executions(compiled_commands, names, programs, self.args.program_report)
+		executables = [os.path.join(self.EXECUTABLES_DIR, self.get_executable(solution))
+							for solution in solutions]
+		compiled_commands = zip(solutions, executables, compilation_results)
+		names = solutions
+		return self.perform_executions(compiled_commands, names, solutions, self.args.program_report)
+
+
+	def print_expected_scores(self, expected_scores):
+		yaml_dict = { "sinol_expected_scores": expected_scores }
+		print(yaml.dump(yaml_dict, default_flow_style=None))
+
+
+	def validate_expected_scores(self, results):
+		new_expected_scores = {} # Expected scores based on results
+
+		for solution in results.keys():
+			new_expected_scores[solution] = {
+				"expected": results[solution],
+				"points": self.calculate_points(results[solution])
+			}
+
+		config_expected_scores = self.config["sinol_expected_scores"] if "sinol_expected_scores" in self.config else {}
+		used_solutions = results.keys() # Solutions that were used
```

Comment:
Non-helpful comment. We can describe it in more detail or we can remove the comment.

Reply:
I removed the comment 37a3d24b153913db62727496c4c3cba3d4dc75ef